### DISCUSSION TITLE
How to order conditionally by either created or updated column

### DISCUSSION THREAD

**Discussion by rilek** [supabase/supabase#8013] [Questions] [answered]
Hi there!
Most of my tables have `created_at` and `updated_at` columns, and `updated_at` is NULL by default. When I am doing regular query I'd like to order them from newest to oldest. How can I do that? With SQL I can simply do following code, but supabase requires to pass single column. Of course I could just fill updated column with current date same as created, but maybe there is another way. 

```sql
select * from [table] order by coalesce(updated_at, created_at) desc
```

**Comment by GaryAustin1** [comment, marked as answer]
That is not possible at the API level.  
You can chain multiple order() but that does not do what you want.
You can post process your resulting table on the client.
You can set updated_at to create_at like you said.
You can use a computed column https://github.com/supabase/supabase/discussions/6005#discussioncomment-2416130
You can use a view.
